velox/vector/fuzzer/VectorFuzzer.h

```diff
@@ -335,10 +335,22 @@ class VectorFuzzer {
   /// pointing to (baseVectorSize-1).
   BufferPtr fuzzIndices(vector_size_t size, vector_size_t baseVectorSize);
 
+  template <typename Class>
+  void registerOpaqueTypeGenerator(
+      std::function<std::shared_ptr<Class>(FuzzerGenerator& rng)> generator) {
+    auto generatorTypeErased = [generator](FuzzerGenerator& rng) {
+      return generator(rng);
+    };
+    opaqueTypeGenerators_[std::type_index(typeid(Class))] = generatorTypeErased;
+  }
+
  private:
   // Generates a flat vector for primitive types.
   VectorPtr fuzzFlatPrimitive(const TypePtr& type, vector_size_t size);
 
+  // Generates a flat vector for opaque types.
```

Comment:
nit: after you add the checks I mentioned earlier, can you also update the comment to mention that it can throw for those cases